From 743130dd2773930b35101e97bcdc8ff1a93680c9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=A9bastien=20Delafond?= <sdelafond@gmail.com>
Date: Thu, 18 May 2017 11:16:03 +0200
Subject: [PATCH 1/8] untangle: bridge mac ageing fix

This patch allows ingress shaping and shaping over multiple
interfaces.  Patch retreived from:

https://github.com/imq/linuximq/blob/master/kernel/v4.x/linux-4.16-imq.diff
---
 net/bridge/br_input.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/net/bridge/br_input.c b/net/bridge/br_input.c
index 59a318b..f63bc49 100644
--- a/net/bridge/br_input.c
+++ b/net/bridge/br_input.c
@@ -283,6 +283,14 @@ static rx_handler_result_t br_handle_frame(struct sk_buff **pskb)
 			goto drop;
 	}
 
+	if (p->state == BR_STATE_LEARNING ||
+            p->state == BR_STATE_FORWARDING) {
+               u16 vid = 0;
+               br_vlan_get_tag(skb, &vid);
+	       br_fdb_update(p->br, p, eth_hdr(skb)->h_source, vid, false);
+        }
+
+
 	if (unlikely(is_link_local_ether_addr(dest))) {
 		u16 fwd_mask = p->br->group_fwd_mask_required;
 
-- 
2.32.0

